GUIDELINES.md

```diff
@@ -0,0 +1,994 @@
+
+#### Using steps classes
+
+To create a steps class you have to derive from the `BaseSteps` class:
+
+```kotlin
+class LoginSteps(testContext: TestContext) : BaseSteps(testContext, appModuleTestingConfiguration)
+```
+
+You have to reference the module testing configuration (in this case `appModuleTestingConfiguration`) of the module the component under test lies in.
+
+To consume a steps class you have to use the `steps` function inside a test or steps class:
+
+```kotlin
+val sut by steps<LoginSteps>()
+```
+
+By using the variable you have access to the members of the steps class. But if you don't need any access to the members and just want the steps class to load you can consider using `requireSteps` in the configuration:
+
+```kotlin
+override fun configure() = super.configure()
+    .requireSteps<LoginSteps>()
+```
+
+### Dependencies
+
+sweetest is tailored for systems where [dependency injection](https://en.wikipedia.org/wiki/Dependency_injection) is used. As you most likely have no DI during unit testing sweetest makes good for that by offering its own simple way of doing it.
+
+In sweetest all dependencies are treated as singletons, so there will be only one instance of a certain type. So it can only be used in these cases. In all other cases you have to fall back to managing object creation on your own. The automated way should cater for the very most cases, though.
+
+#### Dependency modes
+
+For each dependency type there are two possibilities: either it's configured to be mock or real. Unless you tell sweetest otherwise, all dependencies are set to the "mock" mode by default.
+
+##### Mock
+
+If a dependency is set to the "mock" mode, a Mockito mock is created for the dependency.
+
+* The mock is created lazily on demand
+* The same instance is cached when the same type is needed elsewhere
+* The cache is cleared after each test function run
+
+##### Real
+
+What does it mean if you define a type as "real"?
+
+Instance creation is different from "mock" here: the constructor of the _real_ class will be called. In case there are parameters they will be satisfied with arguments by sweetest automatically. All the arguments will be handled as dependencies exactly the same way in a recursive manner until the dependency graph is built up.
+
+Of course dependencies of dependencies can have different modes. E.g. the `LoginViewModel` can have mode "real", but the underlying dependencies can have mode "mock".
+
+##### Configuring and requiring modes
+
+You should configure dependencies mostly in steps classes (but in very rare cases you might also need to do it in test classes). You can do that by calling `requireX` in the overriden `configure` function:
```

Comment:
I think a sentence about the reason why `You should configure dependencies mostly in steps classes` would be beneficial for better understanding.

Reply:
better that way?

```
In order to have a plug-and-play-like experience you should configure dependencies mostly in steps classes (just in very rare cases you might also need to do it in test classes). You can add the dependency configuration by calling `requireX` in the overridden `configure` function:
```

Reply:
Sounds good 👍